### ISSUE TITLE
COCO object detection image size and inference speed

### ISSUE THREAD

**Issue by CamielK** [baaivision/EVA#6]
First of all, thank you for publishing all this hard work! 

After some detectron2 struggles (it has to be build from the EVA fork in /EVA/det/, not from the facebookresearch repo!) I managed to run the COCO evaluation example to reproduce your object detection and instance segmentation results. Amazing stuff!!

My first question is regarding the image size. The detectron2 default image size for COCO is 1024. In your paper you say that you fine-tuned on COCO and LVIS using 1280<sup>2</sup> inputs, but in the code example published here you use a default image size of 1536 ([cascade_mask_rcnn_vitdet_eva_1536.py](https://github.com/baaivision/EVA/blob/master/det/projects/ViTDet/configs/COCO/cascade_mask_rcnn_vitdet_eva_1536.py)). Why did you use 1536 here?

I did some inference experiments and it seems 1280 is indeed an optimal image size for inference:

| **image_size** | **inference speed (3090)**   | **bbox mAP** (only on 100 val2017 images) |
|----------------|------------------------------|-------------------------------------------|
| 1536           | ~2.35 s/iter                 | 67.1                                      |
| 1280           | ~1.05 s/iter                 | 67.4                                      |
| 1024           | ~1.23 s/iter                 | 66.5                                      |
| 612            | ~0.95 s/iter                 | 59.8                                      |

Second question. I used AMP to autocast the torch model to FP16 which already reduced the 1280<sup>2</sup> inference time from 1.05 s/iter to 0.75 s/iter. I am interested in further optimizing the model for faster inference speeds. Do you think it is possible to optimize your model using TensorRT or ONNX?

Thanks again!

**Comment by Yuxin-CV** [comment]
Hi @CamielK, thanks for your interest in EVA and good to see that you have successfully reproduced our results!

Regarding to your questions:
> Q1: Why did you use 1536 here?

A1: For single-scale inference / evaluation, we find using 1536px during inference can achieve a little better result than using 1280px (the training resolution). We also find in some circumstances, using even larger input resolution (_e.g._, 1664px) can further boost the performance.

> Q2: Do you think it is possible to optimize your model using TensorRT or ONNX?

We don't have experience & plan to optimize the speed-accuracy trade-off of EVA. But since EVA det & seg is built upon the widely used Detectron2. Perhaps you may seek for help from the community?

**Comment by Yuxin-CV** [comment]
I believe the issue at hand was addressed, as such I'm closing this. Feel free to ask if you have further questions.

**Comment by h12sawhney** [comment]
Hi,

I am unable to repeat the experiments on Instance Segmentation that you mention.
Some links are stale (e.g. ([cascade_mask_rcnn_vitdet_eva_1536.py](https://github.com/baaivision/EVA/blob/master/det/projects/ViTDet/configs/COCO/cascade_mask_rcnn_vitdet_eva_1536.py))

You mention "(it has to be build from the EVA fork in /EVA/det/, not from the facebookresearch repo!" ? How?

Thanks.

**Comment by Yuxin-CV** [comment]
> Hi,
> 
> I am unable to repeat the experiments on Instance Segmentation that you mention. Some links are stale (e.g. ([cascade_mask_rcnn_vitdet_eva_1536.py](https://github.com/baaivision/EVA/blob/master/det/projects/ViTDet/configs/COCO/cascade_mask_rcnn_vitdet_eva_1536.py))
> 
> You mention "(it has to be build from the EVA fork in /EVA/det/, not from the facebookresearch repo!" ? How?
> 
> Thanks.

All the config related to EVA-01 det are available at: https://github.com/baaivision/EVA/tree/master/EVA-01/det/projects/ViTDet/configs.
